### DISCUSSION TITLE
Timeout modal

### DISCUSSION THREAD

**Discussion by boxadesign** [ONSdigital/design-system#1771] [Feedback]
Use this place to discuss the [timeout modal component](https://ons-design-system.netlify.com/components/timeout-modal/) in the [ONS Design System](https://ons-design-system.netlify.com)

**Comment by Al-Davies** [comment]
**UR findings to be added to DS**

#Time out modal
tested as part of eQ usability testing 4 October 2021

* When faced with the time-out warning, participants knew to click on the continue survey button
* Variations were given in what they thought would be a reasonable amount of time to display this message between 30 seconds and 2 minutes
* When logged out of the system, most participants expected that their answers would be saved for any question they’d completed i.e. where they had clicked on the save and continue button
* If mid-way through completing questions on a page, they would hope their answers would be saved, but wouldn’t really expect it
* All bar 1 of the participant expected to have to enter their login details again once clicking on the sign back in link
* When signing back in, participants expected to be taken back to the survey list page, although 2 would have liked to have gone back to the question they were on when they were timed out. Again, participants had said it was easy enough to get from the survey list page back to the question they were last on
* Participants understood that ‘being inactive’ was that they weren’t actively doing something on the system – the main assumption being because they hadn’t clicked on anything
* None of the participants really understood what ‘followed a link to the middle of the survey’ meant

**Comment by jrbarnes9** [comment]
DAC: Modal heading should be ‘h2’

**Reply by jrbarnes9** [reply]
#2335

**Comment by jrbarnes9** [comment]
Modal not centred in page on mobile
![Screenshot 2022-07-19 at 12 31 36](https://user-images.githubusercontent.com/43346934/179775240-63ce75fc-55b0-44eb-a461-2bf8a56cb16e.png)

**Reply by jrbarnes9** [reply]
#2341

**Comment by jrbarnes9** [comment]
When testing the timeout modal with a screenreader (NVDA) I noticed it by the time it read out `countdownText`, the time announced was ~5 seconds behind the time visible on the page. So the user would be signed out sooner than expected.

Is there a way to ensure the time read out is exactly as displayed, rather than at the end of the long `countdownText` string?

**Reply by jrbarnes9** [reply]
#2442

**Comment by MuslumYilmaz** [comment]
Additional implementation/testing example: I maintain FrontendAtlas, a freemium frontend interview and practice platform. We recently published a free browser-based incident exercise about a modal that opens visually but fails screen-reader users:

https://frontendatlas.com/incidents/modal-screen-reader-failure

The exercise covers moving focus into the modal, trapping and restoring focus, `role="dialog"`, `aria-modal`, accessible naming, and keeping background content inactive. Given the NVDA and timeout-announcement discussion here, would this be useful as a supplementary implementation or testing example in the timeout modal guidance?

This is not ONS-specific user research, so please disregard it if the page should remain limited to validated service research.

Disclosure: I am FrontendAtlas's creator/maintainer. The linked exercise is free; the wider platform also contains paid content. I used AI assistance to help draft this comment. No paid, reciprocal, or special link treatment is requested.
